### PR TITLE
 [l10n] Add Belarusian (be-BY) locale

### PR DESCRIPTION
**Pull request by volhalink**

### CODE REVIEW

**Comment by volhalink** [comment]
Can I use renovate bot somehow like in "Bump MUI Core #7419"? 
Or could anyone update the _mui/material_ to **5.11.5** as it's needed for this PR, please?

**Comment by LukasTy** [comment]
> Or could anyone update the _mui/material_ to **5.11.5** as it's needed for this PR, please?

Why do you think that the library bump is needed?
If you are referring to the same locale released in this version—I do believe that it technically has no relationship with it, except for the fact that it will be nice to have `beBY` locale for both core and `mui-x`, but that depends on the library versions that the user decides to install.

**Comment by volhalink** [comment]
> > Or could anyone update the _mui/material_ to **5.11.5** as it's needed for this PR, please?
> 
> ~Why do you think that the library bump is needed? If you are referring to the same locale released in this version—I do believe that it technically has no relationship with it, except for the fact that it will be nice to have `beBY` locale for both core and `mui-x`, but that depends on the library versions that the user decides to install.~
> 
> Edit: Sorry for the confusion and forgetting the quite recent change in [data grid localization](https://github.com/mui/mui-x/pull/7646/files#diff-1a52cf4cdf6576b7d5a5e3ba04d1135846dab1b00bb57835ae36a7af2906074aR183). 🙈 In that case, let's just wait for dependabot to do it's job, create the PR at the predefined time: _"Schedule: Branch creation - "on sunday before 6:00am" in timezone UTC, Automerge - At any time (no schedule defined)"_, let us handle merging it (if it causes any conflicts) and then sync your branch. 😉

Ok, thank you! :smiley:

**Comment by LukasTy** [comment]
> Ok, thank you! 😃

The `MUI Core` library bump PR has been merged. 😉 
You can sync your PR with latest next. 👍

**Review by MBilalShafi — APPROVED** [review]
Thanks for the contribution @volhalink

**Comment by volhalink** [comment]
> Thanks for the contribution @volhalink

You are welcome :slightly_smiling_face:

**Comment by cherniavskii** [comment]
I thought we'll need to bump [the minimal supported version of `@mui/material` in `peerDependencies`](https://github.com/mui/mui-x/blob/1ac52394955224269f7679b94eff2cc74ad0d73f/packages/grid/x-data-grid/package.json#L56) since there's no `beBY` export in `@mui/material` < 5.11.5.

So I created two sandboxes to test this using the `@mui/x-data-grid` package built from this PR's branch in Codesandbox:
1. With `@mui/material@5.11.5`: https://codesandbox.io/s/customlocaletextgrid-demo-mui-x-forked-9uutn2?file=/package.json
  I expect this one to work since the `beBY` locale is exported in both packages
2. With `@mui/material@5.10.0`: https://codesandbox.io/s/customlocaletextgrid-demo-mui-x-forked-i58v9o?file=/package.json
  I expected this to throw an error since `@mui/x-data-grid` tries to import the `beBY` locale from `@mui/material` but it does not exist in this version of the package. But it works without errors, and the default English locale is used for the pagination component.

I guess the missing named export doesn't throw the error?

**Comment by LukasTy** [comment]
> I thought we'll need to bump [the minimal supported version of `@mui/material` in `peerDependencies`](https://github.com/mui/mui-x/blob/1ac52394955224269f7679b94eff2cc74ad0d73f/packages/grid/x-data-grid/package.json#L56) since there's no `beBY` export in `@mui/material` < 5.11.5.

Great point! 👍 Did not realise this fact... 🤦 
 

> I guess the missing named export doesn't throw the error?

CodeSandbox is always unreliable to test complete scenarios and in this case it once again proves the same.
I did replicate the setup on a local standalone project and got the following error:
Screenshot 2023-01-25 at 22 08 22
Maybe codesandbox "eats-away" errors thrown in `node_modules` or some other environment setup case.

Whatever reason there is for codesandbox to hide the error away it does not change the fact that the final result will not be correct or even error out.
IMHO it seems perfectly reasonable to bump the required `@mui/material` `peerDependencies` version, after all, it's only a minor bump. But maybe there is some sort of a reason why we are not doing it when we need/should? 🤔

**Comment by MBilalShafi** [comment]
I also agree we should have the per-package `peerDependencies` in sync with the version in root `package.json`. 

**Upside:**

Another reason apart from new locales not being available might be that in the dev environment and demos we have later versions, so we can not confirm whether or not the `mui/material` version in `peerDependencies` is still fully compatible with the respective `mui-x` package. (I believe tests are also using the root package versions)

For instance, I tried using the same `@mui/material` version in [this](https://codesandbox.io/s/data-grid-community-forked-gud56z?file=/src/App.tsx) codesandbox (which we have in `peerDependencies` i.e `v5.4.1`) but it threw an error, increasing the version fixed it.

**Downside:**

One downside of bumping the peerDependencies version might be that the users will have to upgrade their local version of `@mui/material`, but that I believe will be handled internally by the package manager. 🤔 

As an experiment, I tried using a previous minor version of `@mui/material` i.e `v5.3.0` (the one in `peerDependencies` is `5.4.1`) with the latest `mui-x` version (v6.0.0-beta.0) in a new CRA app and I was successfully able to run the development server and build the project without errors.

**Conclusion:**

To sum up, IMO, if we bump per-package `peerDependencies` version to the latest one, the users can still keep using the version they have, obviously upgrading to the latest on their part is a plus, but it will not be something that they will be forced to do. (At least if they don't import anything that corresponds to the new versions, like `beBY` locale for instance as @LukasTy pointed out)

**Comment by LukasTy** [comment]
> Another reason apart from new locales available might be that in the dev environment and demos we have later versions, so we can not confirm whether or not the `mui/material` version in `peerDependencies` is still fully compatible with the respective `mui-x` package.

This is a really good point you made, we are manually using one of the later versions and CodeSandbox is using `latest`, hence, the visibility of actual inter-package compatibility with older versions is unknown to us.

> One downside of bumping the peerDependencies version might be that the users will have to upgrade their local version of `@mui/material`, but that I believe should be handled internally by the package manager.

It will probably depend on package manager used. Legacy `yarn` (v1) and older `npm` versions do not handle `peerDependencies` bumping, while never `npm` versions and `pnpm` have different strategies. So, users might end up with an installation error or an automatic dependency bump.

**Comment by cherniavskii** [comment]
> As an experiment, I tried using a previous minor version of @mui/material i.e v5.3.0 (the one in peerDependencies is 5.4.1) with the latest mui-x version (v6.0.0-beta.0) in a new CRA app and I was successfully able to run the development server and build the project without errors.

What version of `@mui/material` was installed in this case?

**Comment by MBilalShafi** [comment]
> What version of @mui/material was installed in this case?

v5.3.0

**Comment by cherniavskii** [comment]
> v5.3.0, at least in node_modules it was this one.

Does it mean that you'll get [an error like this](https://github.com/mui/mui-x/pull/7646#issuecomment-1404299827) if the locale is not available in 5.3.0? Then you'll be forced to upgrade, right?

Given that we only use `MuiTablePagination` from the core locales, I would be in favor of copying those locales from the material package instead of bumping the version and forcing our users to upgrade `@mui/material` just for the locale.

As a user, if I'm not using that specific locale, I shouldn't be forced to upgrade.

**Comment by LukasTy** [comment]
> As a user, if I'm not using that specific locale, I shouldn't be forced to upgrade.

My thoughts exactly, but for this we should have a different approach handling localization.
And your suggestion does honestly sound like a good compromise, given how much is actually re-used by the merging fact. 🙈

**Comment by cherniavskii** [comment]
> My thoughts exactly, but for this we should have a different approach handling localization.

To be clear - we don't have to replace import with a copy of core locales for every locale, but only for those that are not available in the version we have in peerDependencies (so probably only `beBY` and those that will be added in the future)

**Comment by MBilalShafi** [comment]
> Does it mean that you'll get https://github.com/mui/mui-x/pull/7646#issuecomment-1404299827 if the locale is not available in 5.3.0? Then you'll be forced to upgrade, right?

Exactly, my bad. I was trying with the published version and not the current one with the new locale.

I'll create a PR with a copy of two locales for `MuiTablePagination` that are after the current peerDependencies' version. Thank you 🙏

**Comment by cherniavskii** [comment]
> I'll create a PR with a copy of two locales for MuiTablePagination that are after the current peerDependencies' version.

Sounds great!
